var/spack/repos/builtin/packages/openfoam-com/package.py

```diff
@@ -659,6 +663,10 @@ def __init__(self, spec, **kwargs):
         # spec.architecture.target is like `uname -m`
         target   = spec.architecture.target
 
+        # Platform detects as 'cray' - treat like 'linux'
+        if platform == 'cray':
+            platform = 'linux'
```

Comment:
Do you mean you want to keep the `if platform == 'cray'` block at the top, then let the `if platform == 'linux'` block deal with the targets?

That doesn't work on the cray platform, because `spec.architecture.target` isn't set to something like `x86_64` by default.  On my system, it's set to `haswell`, because I have the `craype-haswell` module loaded.  (Take a look at `lib/spack/spack/platforms/cray.py`.)

So this `package.py` ends up trying to modify the wmake rules in `linuxGcc`, but OpenFOAM tries to use the wmake rules in `linux64Gcc`, leading to a bunch of errors when building.

That's why I suggested treating the `platform == 'cray'` as a separate `elif` block at the end, and adding `platform += '64'` based on the `self.arch_option` instead of the `target`, because the `target` can't be trusted on the cray platform.